File: Monster Monster.py
# cook your dish here
test_cases = int(input())

for _ in range(test_cases):
    num_monsters, health_increment = map(int, input().split())
    health_points = list(map(int, input().split()))
    health_points.sort(reverse=True)
    min_attack_stat = 0
    for index in range(num_monsters):
        required_attack = health_points[index] + health_increment * index
        if required_attack > min_attack_stat:
            min_attack_stat = required_attack
    print(min_attack_stat)

File: Monster Monster.cpp
#include <bits/stdc++.h>
using namespace std;


int main() {
    int testCases;
    cin >> testCases;

    while (testCases--) {
        int numMonsters, healthIncrement;
        cin >> numMonsters >> healthIncrement;

        vector<int> healthPoints(numMonsters);
        for (int i = 0; i < numMonsters; ++i) {
            cin >> healthPoints[i];
        }

        // Sort health points in descending order
        sort(healthPoints.rbegin(), healthPoints.rend());

        long long minAttackStat = 0;
        for (int index = 0; index < numMonsters; ++index) {
            long long requiredAttack = healthPoints[index] + static_cast<long long>(healthIncrement) * index;
            if (requiredAttack >= minAttackStat) {
                minAttackStat = requiredAttack;
            }
        }

        cout << minAttackStat << endl;
    }

    return 0;
}
